flybywiresim/a32nx
# refactor(fws): Port PseudoFWC to non-React framework

## Summary of Changes

This PR continues the already existing work on the [pseudo-fwc-rewrite](https://github.com/flybywiresim/a32nx/tree/pseudo-fwc-rewrite) branch to move the PseudoFWC code away from React, due to performance and stuttering issues.

~~Due to the ongoing work with the [Rust FWS](https://github.com/flybywiresim/a32nx/pull/4872), @Benjozork and I briefly discussed and decided to rebase the work onto the experimental branch due to a few FWS features that haven't made it to master yet.~~

~~The plan is to fully port the PseudoFWC to experimental, then make the minor modifications needed to get it compatible with master in a separate PR.~~

After further discussion, we've decided to target master first instead. However, moving to experimental should be as simple as uncommenting the FWC warnings and renaming a couple simvars.

## Screenshots (if necessary)



## References

Existing PseudoFWC code




## Additional context



Discord username (if different from GitHub): Eearslya#7831

## Testing instructions


If everything has been done correctly, this PR should have no visible change to the aircraft at all, other than improved performance and reduced stuttering. Tests should be focused on ensuring that all ECAM memos and warnings appear when they're expected to. Testing should also ensure that warnings can be silenced, cleared, and recalled as appropriate.

1. When loading and starting the aircraft, ensure there are no yellow or red ECAM warnings.
1. Two minutes after both engines have started, ensure the TAKEOFF checklist memo appears.
1. Set up a Flaps 1 takeoff in the MCDU PERF page. Leave flaps at 0. Observe that pressing T.O CONFIG generates a warning.
1. Set flaps 1, ensure correct V-speeds are set in MCDU PERF and press T.O CONFIG again. Observe T.O CONFIG NORMAL memo.
1. Perform a normal takeoff and observe the following:
    - T.O INHIBIT memo appears after 3 seconds of takeoff thrust being applied.
    - T.O INHIBIT memo disappears after reaching approximately 1,500 feet of altitude.
1. In flight, test a few warnings and ensure they function as expected. e.g. Deploy flaps while at speed to create an OVERSPEED warning.
1. Use the Failures menu of the EFB to cause a non-critical failure of a system such as ELAC 1 or SEC 1.
1. Using the CLR and RCL buttons below the ECAM, ensure the warning message can be dismissed and recalled. Disable the failure on the EFB menu.
1. When coming in for landing, the purple LDG INHIBIT memo should appear below 800 feet of altitude.


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
